src/foam/core/CSV.js

```diff
@@ -0,0 +1,382 @@
+
+      return true;
+    },
+
+    function outputHeaderTitle(o) {
+      this.out('"').writePrefix().out(o).out('"').out(this.delimiter);
+    },
+
+    {
+      name: 'outputPropertyFilteredName',
+      code: foam.mmethod({
+        // Is there a `switch` like fall through so the function call isn't unecessarily repeated?
+        Undefined: function(o) { this.outputHeaderTitle(o); },
+        Null:      function(o) { this.outputHeaderTitle(o); },
+        String:    function(o) { this.outputHeaderTitle(o); },
+        Number:    function(o) { this.outputHeaderTitle(o); },
+        Boolean:   function(o) { this.outputHeaderTitle(o); },
+        Date:      function(o) { this.outputHeaderTitle(o); },
+        FObject:   function(o) {
+          // Gets and recurses through object properties
+          var ps = o.cls_.getAxiomsByClass(foam.core.Property);
+
+          for ( var i = 0 ; i < ps.length ; i++ ) {
+            this.outputPropertyName(o, ps[i]);
+          }
+        },
+        Array: function(o) { /* Ignore arrays in CSV */ },
+        Function: function(o) { /* Ignore functions in CSV */ },
+        Object: function(o) {
+          // TODO: How to test?
+          if ( o.outputCSV ) {
+            o.outputCSV(this);
+          } else {
+            if ( this.sortObjectKeys ) {
+              this.outputSortedObjectKeyValues_(o);
+            } else {
+              this.outputObjectKeyValues_(o);
+            }
+          }
+        }
+      })
+    },
+
+    function outputPropertyName(o, p) {
+      if ( ! this.externalProperty(o, p) ) return;
+
+      // Checks if property is object with properties of its own
+      // TODO: Is this correct way to check if object has relevant sub-properties
+      if ( p.of ) {
+        this.start(p.name);
+        this.outputPropertyFilteredName(o[p.name]);
+        this.end();
+      } else {
+        this.outputPropertyFilteredName(p.name);
+      }
+    },
+
+    function outputProperty(o, p) {
+      if ( ! this.externalProperty(o, p) ) return;
+
+      this.output(o[p.name]);
+      this.out(this.delimiter);
+    },
+
+    function reset() {
+      this.buf_ = '';
+      return this;
+    },
+
+    // Starts a nested object, 1 more `__` with object name
+    function start(objectName) {
+      this.nestedObjectNames.push(objectName);
+      return this;
+    },
+
+    function writePrefix() {
+      for ( var i = 0 ; i < this.nestedObjectNames.length ; i++ ) {
+        this.out(this.nestedObjectNames[i] + this.nestedObjectSeperator);
+      }
+
+      return this;
+    },
+
+    // End a nested object, 1 fewer `__` within object name
+    function end() {
+      this.nestedObjectNames.pop();
+      return this;
+    },
+
+    // TODO: Copied from foam.core.json (move to common library)
+    function nl() {
+      if ( this.nlStr && this.nlStr.length ) {
+        this.out(this.nlStr);
+      }
+
+      return this;
+    },
+
+    // TODO: Copied from foam.core.json (move to common library)
```

Comment:
Partially. I removed everything except the escaping of `"`. We still need that as all CSV properties are surrounded by `"` by default (configurable), and it would cause an issue if they are present in the string values.